Open Judge System/Services/OJS.Services/Business/ParticipantScores/ParticipantScoresBusinessService.cs

```diff
@@ -29,5 +34,91 @@ public void RecalculateForParticipantByProblem(int participantId, int problemId)
                 this.participantScoresData.DeleteForParticipantByProblem(participantId, problemId);
             }
         }
+
+        public void NormalizePointsThatExceedAllowedLimitByContest(int contestId)
+        {
+            using (var scope = TransactionsHelper.CreateTransactionScope())
+            {
+                this.InternalNormalizeSubmissionPoints(contestId);
+                this.InternalNormalizeParticipantScorePoints(contestId);
+
+                scope.Complete();
+            }
+        }
+
+        public (int updatedSubmissionsCount, int updatedParticipantScoresCount) NormalizeAllPointsThatExceedAllowedLimit()
+        {
+            int updatedSubmissionsCount,
+                updatedParticipantScoresCount;
+
+            using (var scope = TransactionsHelper.CreateTransactionScope())
+            {
+                updatedSubmissionsCount = this.InternalNormalizeSubmissionPoints();
+                updatedParticipantScoresCount = this.InternalNormalizeParticipantScorePoints();
+
+                scope.Complete();
+            }
+
+            return (updatedSubmissionsCount, updatedParticipantScoresCount);
+        }
+
+        public int InternalNormalizeSubmissionPoints(int? contestId = null)
+        {
+            var updatedSubmissionsCount = 0;
+            IQueryable<Submission> submissionsQuery;
+
+            if (contestId.HasValue)
+            {
+                submissionsQuery = this.submissionsData
+                    .GetAllHavingPointsExceedingLimitByContest(contestId.Value);
+            }
+            else
+            {
+                submissionsQuery = this.submissionsData.GetAllHavingPointsExceedingLimit();
+            }
+
+            var submissions = submissionsQuery.Include(s => s.Problem).ToList();
+
+            foreach (var submission in submissions)
+            {
+                submission.Points = submission.Problem.MaximumPoints;
+
+                this.submissionsData.Update(submission);
+
+                updatedSubmissionsCount++;
+            }
+
+            return updatedSubmissionsCount;
+        }
+
+        public int InternalNormalizeParticipantScorePoints(int? contestId = null)
+        {
+            var updatedParticipantScoresCount = 0;
+            IQueryable<ParticipantScore> participantScoresQuery;
+
+            if (contestId.HasValue)
+            {
+                participantScoresQuery = this.participantScoresData
+                    .GetAllHavingPointsExceedingLimitByContest(contestId.Value);
+            }
+            else
+            {
+                participantScoresQuery = this.participantScoresData.GetAllHavingPointsExceedingLimit();
+            }
+
+            var participantScores = participantScoresQuery.Include(ps => ps.Problem).ToList();
```

Comment:
Is Include necessary here? I can see how it prevents lazy loading queries on every `foreach` iteration, however if `Problem` is big ( or becomes bigger at some point in the future ) it could result in more traffic to include the whole object, while we only need `MaximumPoints`.